### PR TITLE
adding building-packages

### PR DESCRIPTION
**Pull request by KevinSayers**
Adding the docs for building packages. Only modified the Jekyll specific things for the GH pages nav.

### CODE REVIEW

**Comment by KevinSayers** [comment]
@chchang6 I flattened the structure for this so it is now under General. Ready for a review whenever you have a chance.

One other idea I had was we could combine all the markdowns to one, and have a table of contents at the top to jump to different sections. Alternatively, at the bottom of each have a link to the next page. Currently you have to navigate back to the first page to get to the next link.

File: General/intermediate/building-packages/building-packages.md [hunk 1]
```diff
@@ -34,5 +34,5 @@ have been built against a more universally available but lower performance
 library. The same holds for optimizing compilers.
 * Curiosity to know more about the tools you use.
 * Pride of building one's tools oneself.
-* For the sheer thrill of building packages. :smile:
```
Comment (chchang6) [inline General/intermediate/building-packages/building-packages.md]: We have something against smileys?

**Review by chchang6 — APPROVED** [review]
Thanks Kevin, looks fine I think.

**Comment by chchang6** [comment]
Hi Kevin, I merged the PR. Was this on the master branch?

File: General/intermediate/building-packages/building-packages.md (hunk 1 above)
Comment (KevinSayers) [reply General/intermediate/building-packages/building-packages.md]: @chchang6 apparently Jekyll has something against smileys, it left it as plaintext for the GH pages 😆

**Pull request merged**